src/vm/analysis/type_checker/tests/mod.rs

```diff
@@ -2425,3 +2425,79 @@ fn test_string_utf8_negative_len() {
         _ => false,
     });
 }
+
+#[test]
+fn test_parse_principal() {
+    let good = [
+        r#"(parse-principal version 'STB44HYPYAT2BB2QE513NSP81HTMYWBJP02HPGK6)"#,
```

Comment:
A while back, we settled on a C convention for naming our function (map-get, var-set, etc) - the rationale can be found here https://github.com/blockstack/stacks-blockchain/issues/989.
If we're going to stick to our conventions, I think this functions should be named `principal-parse` and `principal-assemble` instead.